connectors/src/connectors/snowflake/lib/snowflake_api.ts

```diff
@@ -0,0 +1,79 @@
+import type { Result } from "@dust-tt/types";
+import type { SnowflakeCredentials } from "@dust-tt/types";
+import { Err, Ok } from "@dust-tt/types";
+import type { Connection, RowStatement, SnowflakeError } from "snowflake-sdk";
+import snowflake from "snowflake-sdk";
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type SnowflakeRows = Array<any> | undefined;
+
+export const testConnection = async ({
+  credentials,
+}: {
+  credentials: SnowflakeCredentials;
+}): Promise<Result<string, Error>> => {
+  const connection = snowflake.createConnection(credentials);
+
+  try {
+    const conn = await connectToSnowflake(connection);
+    const rows = await executeQuery(conn, "SHOW DATABASES");
```

Comment:
As shared on Slack doing the strict minimum here, to be improved later (such as checking we're read-only)

Reply:
should we add a small TODO(SNOWFLAKE) in case we forget ?